### PR TITLE
retries: extract B2RetryPolicy so it can be replaced.

### PR DESCRIPTION
**Pull request by certainmagic**
B2DefaultRetryPolicy implements the default policy which should be fine
for almost all sdk users.  So far, B2StorageClientImpl always uses the
default policy.  I'll work on that in an upcoming change.

Prior to this change, the B2Sleeper was allowed to throw after the
decision was made to sleep some and retry.  As part of extracting the
retry policy, I wanted to be sure that the retry policy would always
find out if the operation succeeded or failed and that wouldn't be the
case if the sleep threw.  I no longer allow the B2Sleeper to throw.
Instead, if it's interrupted, it just resets the thread's interrupted
flag and returns early.  We count on the interruption being noticed
elsewhere (such as in the network call to the B2 servers).

tested with "gradle build javadoc".

### CODE REVIEW

File: core/src/main/java/com/backblaze/b2/client/B2DefaultRetryPolicy.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+package com.backblaze.b2.client;
+
+import com.backblaze.b2.client.exceptions.B2Exception;
+
+/**
+ * B2DefaultRetryPolicy implements the retry policy described in the B2
+ * documentation.  It should be reasonable and sufficient for almost every
+ * use of B2.
+ *
+ * Each attempted operation should have a unique instead of this class
```
Comment (bwbeach) [inline core/src/main/java/com/backblaze/b2/client/B2DefaultRetryPolicy.java]: s/instead/instance/

File: core/src/main/java/com/backblaze/b2/client/B2LargeFileUploader.java [hunk 2]
```diff
@@ -225,23 +226,25 @@ private B2FileVersion uploadPartsAndFinish(B2FileVersion largeFileVersion,
...
+                    final B2UploadPartRequest partRequest = B2UploadPartRequest
+                            .builder(partSpec.partNumber,
+                                    new B2PartOfContentSource(request.getContentSource(), partSpec.start, partSpec.length))
+                            .build();
+
+                    final B2Part part = webifier.uploadPart(uploadPartUrlResponse, partRequest);
+                    uploadPartUrlCache.unget(uploadPartUrlResponse);
+                    return part;
+                },
+                new B2DefaultRetryPolicy());
```
Comment (bwbeach) [inline core/src/main/java/com/backblaze/b2/client/B2LargeFileUploader.java]: Should this use the retry policy supplier?

File: core/src/main/java/com/backblaze/b2/client/B2DefaultRetryPolicy.java (hunk 1 above)
Comment (certainmagic) [inline core/src/main/java/com/backblaze/b2/client/B2DefaultRetryPolicy.java]: fixed locally.

File: core/src/main/java/com/backblaze/b2/client/B2LargeFileUploader.java (hunk 2 above)
Comment (certainmagic) [inline core/src/main/java/com/backblaze/b2/client/B2LargeFileUploader.java]: yes.  good eye.  :)

right now, we always use the default retryer.  in an upcoming change, i'll make it configurable from the outside and i'll fix it here too.

**Pull request merged**